app/controllers/locations_controller.rb

```diff
@@ -155,4 +156,14 @@ def authorise_ip_actions
       redirect_to ips_path
     end
   end
+
+  def check_user_organisations
+    unless current_user.is_super_admin?
+      organisation = Organisation.find(params.dig(:location, :organisation_id))
+      unless current_user.organisations.include? organisation
+        flash[:alert] = "You can't update the location to an organisation you're not a member of"
+        render :edit
```

Comment:
This should redirect because otherwise it won't have a location_id set. Not being a member of the organisation you want to move a location to is also not part of the normal flow so can be handled a bit abruptly